packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -136,25 +136,29 @@ export class ApolloServerBase<Request = RequestInit> {
       this.createSubscriptionServer(this.http, config);
     }
 
-    if (opts.engine || opts.engineInRequestPath) this.createEngine(opts);
+    if (opts.engineProxy || opts.engineInRequestPath) this.createEngine(opts);
 
     return new Promise((success, fail) => {
-      if (this.engine) {
-        this.engine.listen(
-          Object.assign({}, options.engineLauncherOptions, {
-            graphqlPaths: [this.graphqlPath],
-            port: options.port,
-            httpServer: this.http,
-          }),
+      if (this.engineProxy) {
+        this.engineProxy.listen(
+          Object.assign(
```

Comment:
`Object.assign` with an empty object literal and another object literal is just the second object literal :)